compiler/backend/ccgexprs.nim

```diff
@@ -1763,6 +1760,15 @@ proc genMagicExpr(p: BProc, e: CgNode, d: var TLoc, op: TMagic) =
     linefmt(p, cpsStmts, "$1 = ($2)($3);$n", [a.r, typ, rdLoc(b)])
   of mChckRange:
     genRangeChck(p, e, d)
+  of mChckNaN:
+    var a: TLoc
+    initLocExpr(p, e[1], a)
+    # XXX: using a comparison could result in a floating point exception to
+    #      be raised (i.e., the flag to be set in the fenv).
+    #      ``__builtin_isnan``, which doesn't exhibit this behaviour, could be
```

Comment:
Any reason not to do this now?

Reply:
It felt like something unrelated to the rest of the PR, but you're right, there's not really a reason against it -- I'll change it here.

Reply:
I've only changed the comment for now. The `__builtin_isnan` built-in is only available for GCC, Clang, and some other compilers aiming for GCC compatibility, but not for MSVC, and the only portable way of testing for NaNs that I've found would be inspecting the bit pattern (the C `isnan` function only became standard with C99).

Since `mChckNaN` is only intended for testing values resulting from arithmetic operations (which can never be signaling NaNs), I've changed the `mChckNaN` documentation to mention that signaling NaN behaviour is undefined.